### DISCUSSION TITLE
I'm finding to way to encode parameter using viem before writing to contract.

### DISCUSSION THREAD

**Discussion by Asaki-Katsu** [wevm/viem#1616] [Question] [answered]
Hi, I have these 3 structs:
```solidity
struct T1 {
 uint8  foo,
 bytes bar
}
```

However, `bar` is a bytes data that is supposed to be decoded into struct `T2` like this
```solidity
struct T2 {
   T3 foo2,
   bytes bar2
}

struct T3 {
  bytes32 aa,
  address bb,
  uint256 cc,
  uint64 dd
}
```

Could anyone give me a how-to encode this kind of struct with viem in .js before writing to the contract?

**Comment by tmm** [comment]
Can you share some code of what you tried already?

**Reply by Asaki-Katsu** [reply, marked as answer]
Hi, I did solve it, I just simply ignore the 2nd struct like this

```javaScript
const encoded = encodeAbiParameters(
        parseAbiParameters('bytes32 aa, address bb, uint256 cc, uint64 dd, bytes bar2'),
        [
          data1,
          data2,
          data3,
          data4,
          data5,
        ],
      )
```

**Reply by Asaki-Katsu** [reply]
At first I thought I have to encode struct `T3` first, then encode it with `bytes bar2` in order to fit with struct T2. However, tried some and found that just ignore the struct, and encoding it orderly works.

**Reply by Asaki-Katsu** [reply]
Found that by inserting struct in the struct, it's simply like concat the data in the bytes, nothing complicated, I forgot this point haha.
